testing/scenario_app/ios/Scenarios/ScenariosTests/FlutterViewControllerTest.m

```diff
@@ -46,4 +56,25 @@ - (void)testFirstFrameCallback {
   XCTAssertGreaterThan(countDownMs, 0);
 }
 
+- (void)testFirstFrameDidRender {
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"test" project:nil];
+  [engine runWithEntrypoint:nil];
+  FlutterViewControllerFirstFrameOverride* flutterViewController =
+      [[FlutterViewControllerFirstFrameOverride alloc] initWithEngine:engine
+                                                              nibName:nil
+                                                               bundle:nil];
+  self.flutterViewController = flutterViewController;
+
+  AppDelegate* appDelegate = (AppDelegate*)UIApplication.sharedApplication.delegate;
+  UIViewController* rootVC = appDelegate.window.rootViewController;
+  [rootVC presentViewController:self.flutterViewController animated:NO completion:nil];
+  NSRunLoop* runLoop = [NSRunLoop currentRunLoop];
+  int countDownMs = 2000;
+  while (!flutterViewController.firstFrameRendered && countDownMs > 0) {
```

Comment:
You should almost never have to manually spin the main run loop in tests.  The above testFirstFrameCallback test should be:
```
XCTestExpectation *callbackExpecation = [self expectationWithDescription:@"callback"];
[self.flutterViewController setFlutterViewDidRenderCallback:^{
  [callbackExpecation fulfill];
}];
...
[rootVC presentViewController:self.flutterViewController animated:NO completion:nil];
[self waitForExpectationsWithTimeout:2.0 handler:nil];
```
For this one you can use a XCTKVOExpectation.  Something like (I didn't test this):
```
XCTestExpectation *firstFrameRenderedExpecation = [self keyValueObservingExpectationForObject:flutterViewController keyPath:@"setFirstFrameRendered" expectedValue:@YES];
...
[rootVC presentViewController:self.flutterViewController animated:NO completion:nil];
[self waitForExpectationsWithTimeout:2.0 handler:nil];
```
or
```
XCTestExpectation *firstFrameRenderedExpecation = [self keyValueObservingExpectationForObject:flutterViewController keyPath:@"setFirstFrameRendered" expectedValue:@YES];
...
[rootVC presentViewController:self.flutterViewController animated:NO completion:nil];
[self waitForExpectationsWithTimeout:2.0 handler:nil];
```

Or better yet, can you get rid of `firstFrameDidRender` and `setFirstFrameRendered` if it's just for this purpose, and fulfill the expectation however _flutterViewRenderedCallback is set (I don't know how _flutterViewRenderedCallback works).

Reply:
What is this actually testing that's different than the above test? It looks like they are both checking that the _flutterViewRenderedCallback was called.